morph_tool/spatial.py

```diff
@@ -27,3 +34,47 @@ def point_to_section_segment(neuron, point):
             return (section.id + 1, offset[0][0])
 
     raise ValueError('Cannot find point in morphology that matches: {}'.format(point))
+
+
+def align_pyr_morph(morph, direction=None):
+    """In place alignment of a pyramidal morphology such that apical tree is along 'direction'.
+
+    The algorithm is based on SVD decomposititon of the correlation matrix obtained from all points
+    in apical neurites, giving the principal axis of apical dendrite.
+    If not apicals are present, no rotation is applied.
+
+    Args:
+        morph (morphio.Morphology): morphology to align
+        direction (ndarray): 3 vector with for final direction, if None, [0, 1, 0] will be used
+    """
+    if direction is None:
+        direction = [0.0, 1.0, 0.0]
+    apical_points = []
+    for section in morph.iter():
+        if section.type == SectionType.apical_dendrite:
+            apical_points += list(section.points)
```

Comment:
I think it would be more efficient to only iterate over the root sections in order to find the apical root section and then only iterate over the apical tree.